main.tf

```diff
@@ -1,3 +1,25 @@
+locals {
+  listener_endpoints = flatten([for lk, lv in var.listeners : [
+    for ek, ev in lv.endpoint_groups : {
+      listerner_name = lk
+      endpoint_group = ek
+
+      endpoint_group_region         = try(ev.endpoint_group_region, null)
```

Comment:
Instead of copying values from the `ev` one by one, you can just `merge({listerner_name = lk, endpoint_group = ek}, ev)` and keep using `try()` in the `resource "aws_globalaccelerator_endpoint_group" "this" {}`